ra/ra.go

```diff
@@ -468,8 +468,11 @@ func (ra *RegistrationAuthorityImpl) validateContacts(ctx context.Context, conta
 		if parsed.Scheme != "mailto" {
 			return berrors.InvalidEmailError("contact method %q is not supported", parsed.Scheme)
 		}
-		if parsed.RawQuery != "" {
-			return berrors.InvalidEmailError("contact email [%q] contains hfields", contact)
+		if parsed.RawQuery != "" || contact[len(contact)-1] == '?' {
+			return berrors.InvalidEmailError("contact email %q has a question mark in it", contact)
+		}
+		if parsed.Fragment != "" {
+			return berrors.InvalidEmailError("contact email %q has a an invalid character", contact)
```

Comment:
sorry this is super nit-picky, but can we unify these error messages? That's part of what I was doing with the "contains..." suggestion, but even sticking with "has...", can they either both end with "in it" or _neither_ end with that prepositional phrase?

Also, the latter of these could explicitly mention "#", rather than just saying "invalid character", to provide equal levels of detail.